From: Neil Jerram <neil@ossau.uklinux.net>
Date: Sat, 12 Jul 2008 18:22:59 +0000 (+0100)
Subject: Improved MIPS/Linux gc_os_dep.c definitions
X-Git-Url: http://git.savannah.gnu.org/gitweb/?p=guile.git;a=commitdiff_plain;h=4ff3575c77f95ed5c86970fa93629ba3cfab1912;hp=10a529b0273a7bcad314885db4567aea8b39025e

Improved MIPS/Linux gc_os_dep.c definitions

	From Thiemo Seufer <ths@networkno.de>:

	* gc_os_dep.c (CPP_WORDSZ, ALIGN_DOUBLE, DATAEND,
	DYNAMIC_LOADING): Added #defines.
	(_fdata, _end): Added declarations.
	(DATASTART): Use _fdata instead of __data_start.
	(STACKBOTTOM): Changed from 0x80000000 to 0x7fff8000.

Index: sid/THANKS
===================================================================
--- sid.orig/THANKS
+++ sid/THANKS
@@ -77,6 +77,7 @@
          Werner Scheinast
 	   Bill Schottstaedt
 	  Frank Schwidom
+         Thiemo Seufer
           Scott Shedden
            Alex Shinn
          Daniel Skarda
Index: sid/libguile/gc_os_dep.c
===================================================================
--- sid.orig/libguile/gc_os_dep.c
+++ sid/libguile/gc_os_dep.c
@@ -1008,19 +1008,18 @@
 
 # ifdef MIPS
 #   define MACH_TYPE "MIPS"
-/* #   define STACKBOTTOM ((ptr_t)0x7fff8000)  sometimes also works.  */
 #   ifdef LINUX
-      /* This was developed for a linuxce style platform.  Probably	*/
-      /* needs to be tweaked for workstation class machines.		*/
-#     define OS_TYPE "LINUX"
-      extern int __data_start;
-#     define DATASTART ((ptr_t)(&__data_start))
-#     define ALIGNMENT 4
-#     define USE_GENERIC_PUSH_REGS 1
-#     define STACKBOTTOM 0x80000000
-	/* In many cases, this should probably use LINUX_STACKBOTTOM 	*/
-	/* instead. But some kernel versions seem to give the wrong	*/
-	/* value from /proc.						*/
+#       define CPP_WORDSZ _MIPS_SZPTR
+#       define OS_TYPE "LINUX"
+#       define ALIGNMENT 4
+#       define ALIGN_DOUBLE
+        extern int _fdata;
+#       define DATASTART ((ptr_t)(&_fdata))
+        extern int _end;
+#       define DATAEND ((ptr_t)(&_end))
+#       define STACKBOTTOM ((ptr_t)0x7fff8000)
+#       define USE_GENERIC_PUSH_REGS 1
+#       define DYNAMIC_LOADING
 #   endif /* Linux */
 #   ifdef ULTRIX
 #	define HEURISTIC2
